lib/cloudinary/uploader.rb

```diff
@@ -206,17 +206,17 @@ def self.text(text, options={})
     end
   end
 
-  def self.generate_sprite(tag, options={})
+  # Generates sprites by merging multiple images into a single large image.
+  #
+  # @param [Array<String>|String] source Array of urls or a tag
+  # @param [Hash] options Additional options
+  #
+  # @return [Hash] Hash with meta information URLs of generated sprite resources
+  def self.generate_sprite(source, options = {})
```

Comment:
Let's avoid overloading the first parameter based on type (array vs string). With this approach we will easily get in to trouble when this function will get a list of tags (which it actually already supports), public ids, etc.

I suggest to consume `urls` from `options` and make `tag` parameter optional.

This is relevant for `multi` as well.

Reply:
What is a proposed usage of this method? If a client wants to generate a sprite from urls he presumably will call it like `generate_sprite(:urls => [url1, url2])` but in this case `tag` variable (in this very case `source`) will be set to hash which again brings us to overloading first parameter but now it will be `Hash` vs `String`. 

We either have to call this method `generate_sprite(nil, :urls => [url1, url2])` (and presumably leave first parameter mandatory) or do something like this inside the method (which of course might be extracted to a helper method):

```ruby
if tag.is_a?(Hash)
  if options.blank?
    options = tag
    tag = nil
  else
    raise "First argument must be a tag when additional options are passed"
  end
end
urls = options.delete[:urls]
```

Reply:
@YomesInc , let's go with the approach (code) you suggested, and make `tag` parameter optional. You can implement this only once in `build_multi_and_sprite_params` function